catroid/src/main/java/org/catrobat/catroid/ui/fragment/FormulaEditorFragment.java

```diff
@@ -857,4 +850,22 @@ public void updateButtonsOnKeyboardAndInvalidateOptionsMenu() {
 			backspaceOnKeyboard.setEnabled(true);
 		}
 	}
+
+	private boolean discardFormulaIfPossible() {
```

Comment:
wouldnt it be easier to save the initial state of a formula instead of unrolling the complete undo stack?

also, dont call this discardFormulaIfPossible, because this it performs the undo if theres unsaved changes. therefore when it is "nessecary", not possible, and it performs this on FormulaS.
but anyways, dont rename this method, but split it, and have the check if changes are to be discarded where it is called, and only make a discardChanges method.

Reply:
no because in case of colorseekbar you have 3 states which should be saved so it is easier to iterate through the stack